### ISSUE TITLE
Section headers without following TextPart should act as closing tags for the section.

### ISSUE THREAD

**Issue by raffazizzi** [OpenITI/oitei#7]
This was originally reported as an issue for oimpd, but it should be addressed in oitei.

Original issue: https://github.com/OpenITI/oimdp/issues/6

> OpenITI mARkdown does generally not contain closing tags, but sometimes we do need to be able to indicate that a subsection is included in the text of a section, and that the text of the main section continues after the end of the subsection.
> 
> We do this by repeating the tag of the subsection, but without a header (only a space after the pipes, like this:
> 
> ```
> ### | Section 1
> # intro
> ### || Section 2
> # short subsection
> ### || 
> # Section 1 continues after end of section 2
> ```
> 
> Ideally, this should be converted to:
> 
> ```
>       <div>
>         <head>Section 1</head>
>         <p>
>           <lb/>intro
>         </p>
>         <div>
>           <head>Section 2</head>
>           <p>
>             <lb/>short subsection
>            </p>
>         </div>
>         <p>
>           <lb/>Section 1 continues after end of section 2
>         </p>        
>       </div>
> ```
> 
> Currently it is converted to:
> 
> ```
>       <div>
>           <head>Section 1</head>
>           <p>
>               <lb/>intro
>           </p>
>           <div>
>               <head>Section 2</head>
>               <p>
>                   <lb/>short subsection
>               </p>
>           </div>
>           <div>
>              <head></head>
>               <p>
>                   <lb/>Section 1 continues after end of section 2
>               </p>
>           </div>      
>       </div>
> ```

**Comment by raffazizzi** [comment]
@pverkind 

I started working on this, but I just realized that this is not legal in TEI: sections cannot stop and resume; when one finishes another one starts. What I can do is to provide a new sibling section like this. Would this make sense for the texts you're working with?

```
 <div>
    <head>Section 1</head>
    <p>
        <lb/>intro
    </p>
    <div>
        <head>Section 2</head>
        <p>
           <lb/>short subsection
        </p>
    </div>
    <div>
        <p>
          <lb/>Section 1 continues after end of section 2
        </p>
    </div>
</div>
```

**Comment by pverkind** [comment]
Dear Raff, I may misunderstand the TEI rules, but section 1 does not really end with the start of section 2, right? Section 2 is merely a nested subsection of section one. 

If even that is illegal in TEI, perhaps there's another solution. The TEI documentation contains a similar example (https://tei-c.org/release/doc/tei-p5-doc/en/html/examples-floatingText.html): 

```
<body>
 <div type="scene">
  <sp>
   <p>Hush, the players begin...</p>
  </sp>
  <floatingText type="pwp">
   <body>
    <div type="act">
     <sp>
      <l>In Athens our tale takes place [...]</l>
     </sp>
<!-- ... rest of nested act here -->
    </div>
   </body>
  </floatingText>
  <sp>
   <p>Now that the play is finished ...</p>
  </sp>
 </div>
</body>
```

It uses the floatingText tag to enclose the subsection: https://tei-c.org/release/doc/tei-p5-doc/en/html/DS.html#DSFLT

This may perhaps be a solution?

**Comment by raffazizzi** [comment]
@pverkind 

> but section 1 does not really end with the start of section 2, right?

That's right, section 2 is a subsection, but in TEI it can only be followed by other subsections, not a return to the parent section. TEI was originally designed around western books, and it still keeps some of those vestiges (e.g. front, body, back). This is another case where if you go from chapter 1 to section 1.1, you can't go back to chapter 1, only to section 1.2, or another (un)named subsection. 

Would you be able to point me to an example of a text in which this kind of encoding is needed (ie where the parent section resumes after a subsection)?

`<floatingText>` is a typically used to bring in texts that are conceptually separate from their surrounding ones, which doesn't seem appropriate for a subsection, since a subsection is part of the main text.
